src/util/expr_util.cpp

```diff
@@ -321,8 +321,19 @@ exprt is_not_zero(
   irep_idt id=
     src_type.id()==ID_floatbv?ID_ieee_float_notequal:ID_notequal;
 
-  exprt zero=gen_zero(src_type);
-  assert(zero.is_not_nil());
+  exprt zero;
+  if(src_type.id()==ID_floatbv ||
+     src_type.id()==ID_fixedbv)
+  {
+    const floatbv_typet &type=to_floatbv_type(src_type);
```

Comment:
Wouldn't it be cleaner to go through the ieee_floatt::make_zero().to_expr() route?